src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -175,19 +177,37 @@ public void sendEmailToRecoverPassword(String email) {
   }
 
   private User findUserByEmailOrThrowException(String email) {
-    return userRepository.findByEmailAndDeletedIsFalse(email).orElseThrow(() -> new RuntimeException("User not found"));
+    return userRepository.findByEmailAndDeletedIsFalse(email).orElseThrow(() -> new RuntimeException("User with email " + email +" does not exists."));
```

Comment:
Cambiemos este tipo de error a un ResponseStatusException que nos permite además modificar el HTTP Status

```suggestion
    return userRepository.findByEmailAndDeletedIsFalse(email).orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "User with email " + email +" does not exists."));
```